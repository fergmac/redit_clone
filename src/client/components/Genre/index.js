import React, { PropTypes } from 'react';
import Divider from 'material-ui/Divider';
import styles from './style.css';
import { List, ListItem } from 'material-ui/List';
import Subheader from 'material-ui/Subheader';
import { Link } from 'react-router';
// was weeks
const Genre = ({ genre }) => (

  <div className={styles.genre}>
    <Subheader>{genre.title}</Subheader>
    <Divider />
    <List>
      {genre.albums.map((album) => (
        <Link to={`/albums/${album.id}`} key={album.id}>
          <ListItem className={styles.links}>
            {album.name}
          </ListItem>
        </Link>))}
    </List>
  </div>
);

Genre.propTypes = {
  genre: PropTypes.object.isRequired,
};

export default Genre;
