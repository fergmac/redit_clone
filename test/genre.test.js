/* eslint-disable */
import React, { PropTypes } from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Genre from '../src/client/components/Genre';


describe('Genre', () => {
    const genreData = {
        id: 1,
        title: 'Jazz',
        artists: [{ id: 1, name: 'Eddie Lang'}]
    }
    it('renders the genres', () => {
       const wrapper = shallow(
                <Genre 
                    genre={genreData}
                />
       )
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })
})


// describe(?, () => {


//     it('', () => {
//         const ? = ()
//         ? = jest.fn()
//         ?()

//         expect().toBeCalled()
//     })
 
// })