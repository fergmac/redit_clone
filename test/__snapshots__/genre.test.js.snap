exports[`Genre renders the genres 1`] = `
<div>
  <Subheader
    inset={false}>
    Jazz
  </Subheader>
  <Divider
    inset={false} />
  <List>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/albums/1">
      <ListItem
        autoGenerateNestedIndicator={true}
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryTogglesNestedList={false}
        secondaryTextLines={1}>
        Eddie Lang
      </ListItem>
    </Link>
  </List>
</div>
`;
